src/common/meta/src/wal/kafka/topic_manager.rs

```diff
@@ -0,0 +1,147 @@
+
+pub type Topic = String;
+
+const METASRV_CREATED_TOPICS_KEY: &str = "metasrv_created_topics";
+const CREATE_TOPIC_TIMEOUT: i32 = 5_000; // 5,000 ms.
+
+pub struct TopicManager {
+    topic_pool: Vec<Topic>,
+    topic_selector: TopicSelectorRef,
+}
+
+impl TopicManager {
+    pub async fn try_new(
+        kafka_opts: Option<&KafkaOptions>,
+        kv_backend: &KvBackendRef,
+    ) -> Result<Self> {
+        let opts = kafka_opts.context(MissingKafkaOptsSnafu)?;
+        let topic_pool = build_topic_pool(opts, kv_backend).await?;
+        let topic_selector = build_topic_selector(&opts.selector_type);
+
+        // The cursor in the round-robin selector is not persisted which may break the round-robin strategy cross crashes.
+        // Introduces a shuffling may help mitigate this issue.
+        let topic_pool = match opts.selector_type {
+            TopicSelectorType::RoundRobin => shuffle_topic_pool(topic_pool),
+        };
+
+        Ok(Self {
+            topic_pool,
+            topic_selector,
+        })
+    }
+
+    pub fn select_topics(&self, num_topics: usize) -> Vec<Topic> {
+        (0..num_topics)
+            .map(|_| self.topic_selector.select(&self.topic_pool))
+            .collect()
+    }
+}
+
+async fn build_topic_pool(opts: &KafkaOptions, kv_backend: &KvBackendRef) -> Result<Vec<Topic>> {
+    let num_topics = opts.num_topics;
+    ensure!(num_topics > 0, InvalidNumTopicsSnafu { num_topics });
+
+    let broker_endpoints = opts.broker_endpoints.clone();
+    let kafka_client = ClientBuilder::new(broker_endpoints.clone())
+        .build()
+        .await
+        .context(BuildKafkaClientSnafu { broker_endpoints })?;
+
+    let kafka_ctrl_client = kafka_client
+        .controller_client()
+        .context(BuildKafkaCtrlClientSnafu)?;
+
+    let topics = (0..num_topics)
+        .map(|topic_id| format!("{}_{topic_id}", opts.topic_name_prefix))
+        .collect::<Vec<_>>();
+
+    let created_topics = restore_created_topics(kv_backend)
+        .await?
+        .into_iter()
+        .collect::<HashSet<Topic>>();
+
+    let num_created_topics = created_topics.len();
+    ensure!(
+        num_created_topics <= num_topics,
+        TooManyCreatedKafkaTopicsSnafu { num_created_topics }
+    );
+
+    let create_topic_tasks = topics
+        .iter()
+        .filter_map(|topic| {
+            if created_topics.contains(topic) {
+                return None;
+            }
+
+            // TODO(niebayes): Determine how rskafka handles an already-exist topic. Check if an error would be raised.
+            Some(kafka_ctrl_client.create_topic(
+                topic,
+                opts.num_partitions,
+                opts.replication_factor,
+                CREATE_TOPIC_TIMEOUT,
+            ))
+        })
+        .collect::<Vec<_>>();
+
+    futures::future::try_join_all(create_topic_tasks)
+        .await
+        .context(CreateKafkaTopicSnafu)?;
+
+    persist_created_topics(&topics, kv_backend).await?;
```

Comment:
I think we should add some info logs for creating topics.

Reply:
Will do so.